scripts/drive_paged_programs.py

```diff
@@ -117,6 +190,164 @@
     "paged_fp8": "spyre_paged_attn_fp8",
 }
 ATTN_NAME = attention_map[args.attention_type]
+CPU_DTYPE = "fp8" if is_fp8 else "fp32"
+
+torch.manual_seed(42)
+torch.set_grad_enabled(False)
+os.environ["COMPILATION_MODE"] = "offline_decoder"
+os.environ["DT_PROG_CRITERIA_FILEPATH"] = args.program_criteria_json_path
+if (
+    "VLLM_DT_MAX_CONTEXT_LEN" not in os.environ
+    or "VLLM_DT_MAX_BATCH_SIZE" not in os.environ
+):
+    if local_rank == 0:
+        dprint(
+            "Please specify VLLM_DT_MAX_CONTEXT_LEN and VLLM_DT_MAX_BATCH_SIZE environment variables"
+        )
+    exit()
+
+max_batch_size = int(os.environ["VLLM_DT_MAX_BATCH_SIZE"])
+max_tkv = int(os.environ["VLLM_DT_MAX_CONTEXT_LEN"])
+
+
+def __prepare_inputs(batch_size, seq_length, tokenizer, enforce_sizes=[], seed=0):
+    start = time.time()
+    prompts_and_sizes = sampler(
+        DATASET_PATH,
+        batch_size,
+        tokenizer,
+        32,
+        seq_length * 2 if allow_truncation else seq_length,
+        seed,
+        enforce_sizes=enforce_sizes,
+        truncation=allow_truncation,
+    )
+    end = time.time()
+    if local_rank == 0:
+        dprint(f"extracted prompts in {(end - start):.4f} seconds")
+    prompt_list = []
+    for prompt, size in prompts_and_sizes:
+        encoded = tokenizer.encode(prompt, return_tensors="pt").squeeze(0)
```

Comment:
nit: given we already have to encode to sample, wouldn't it be best to simply return the encoded samples from the sampler directly?